handbook/engineering/web/index.md

```diff
@@ -13,7 +13,20 @@ The web team's current focus is documented in [the tracking issue for the curren
 
 ## Goals
 
-TODO
+### Long term
+
+***Deliver our users the full, unique value of [extensions](https://docs.sourcegraph.com/extensions).***
+
+**Outcome**: Our webapp, browser extensions and native integrations are great platforms to provide our users unique value through extensions. These platforms are well-maintained, consistent, easy to setup, well-documented, well-tested, performant, and show their power through convincing extensions built on top of them. The extensions platform and API provide powerful capabilities to extension developers and a great developer experience.
+
+### Medium term
```

Comment:
(as discussed in web sync) In web sync you gave extra color/info on what these goals mean, and I think that would be valuable to capture here.

Reply:
I added extra information to each point while trying to balance it with keeping these medium-term goals high-level (as opposed to a sequence of short-term goals). Please let me know what you think!